### ISSUE TITLE
unbalanced-tuple-unpacking when using [].append

### ISSUE THREAD

**Issue by cocos56** [PyCQA/pylint#2461]
This bug appeared when I was writing Python code with VS Code. There is no problem with the code I write, but VScode suggests a problem. As shown in the following picture. You can email me if you need to.(zj175@139. com) I reported this to VS Code, but they said Pylint produced the error, and I should submit a bug report to the Pylint project.

[test.zip](https://github.com/PyCQA/pylint/files/2339116/test.zip)
![image](https://user-images.githubusercontent.com/36597804/44898264-f7aac700-ad30-11e8-9227-2f5ee4412c08.png)
![image](https://user-images.githubusercontent.com/36597804/44898273-fda0a800-ad30-11e8-843b-0ce8b7fba75c.png)

**Comment by PCManticore** [comment]
Thank you for the report!

**Comment by arvindsree** [comment]
Something similar. Looks like it can't handle dynamic lists. Would be really nice if this is supported as this is the only false positive pylint error in our entire code base.

```
def my_func(numbers):
    number_out = []
    for number in numbers:
        number_out.append(number * 2)
    return number_out


def main():
    [_a, _b, _c] = my_func([1, 2, 3])

main()

```

E: 12, 4: Possible unbalanced tuple unpacking with sequence defined at line 4: left side has 3 label(s), right side has 0 value(s) (unbalanced-tuple-unpacking)

**Comment by mbyrnepr2** [comment]
I can't reproduce on pylint version 2.13; perhaps this issue can be closed.

**Comment by DanielNoord** [comment]
Thanks @mbyrnepr2!
